app/Models/WorkshopBalance.php

```diff
@@ -28,56 +25,64 @@ public function semester()
         return $this->belongsTo('App\Models\Semester');
     }
 
-    public function membersPayedKKTNetreg()
-    {
-        return $this->payedKKTNetregInSemester(Semester::current());
-    }
-
-    public function membersPayedKKTNetregInSemester(Semester $semester)
-    {
-        return $this->workshop->users->filter(function ($user, $key) use ($semester) {
-            return $user->isActiveIn($semester) && (! $user->hasToPayKKTNetregInSemester($semester));
-        });
-    }
-
     /**
      * Generates all the workshops' allocated balance in the current semester.
-     * For all active members in a workshop who payed kkt: 
+     * For all active members in a workshop who payed kkt:
      *      kkt * (isResident ? 0.6 : 0.45) / member's workshops' count
     */
-    public static function generateBalances()
+    public static function generateBalances($semester_id)
     {
-        foreach(Workshop::all() as $workshop)
+        $workshops = Workshop::with('users:id')->get();
+
+        if(!self::where('semester_id', $semester_id)->count()) {
+            $balances = [];
+            foreach ($workshops as $workshop) {
+                $balances[] = [
+                    'semester_id' => $semester_id,
+                    'workshop_id' => $workshop->id
+                ];
+            }
+
+            self::insert($balances);
```

Comment:
This line fails with `Field 'resident' doesn't have a default value`.

Reply:
I'll take a look at it this week!

Reply:
Sorry for being slow, I fixed the issue!